### PR TITLE
Feature/image learner split

### PR DESCRIPTION
**Pull request by paulocilasjr**
This PR fixes per‑class split drift in the Image Learner by replacing the two‑stage train_test_split with deterministic per‑label allocation. Each label now receives exact train/validation/test counts derived from the requested split proportions using largest‑remainder rounding, so classes with the same totals no longer end up with off‑by‑one differences. 

We retain the existing fallback to a random split when a class has too few samples for stratification.

### CODE REVIEW

File: tools/imagelearner/image_learner.xml [hunk 1]
```diff
@@ -130,7 +130,9 @@
                 </param>
             </when>
             <when value="auto">
-                <!-- No validation metric selection; tool will infer task and metric. -->
+                <param name="validation_metric_auto" type="select" optional="true" label="Validation metric (auto)" help="Metrics compatible with binary, multi-class, and regression tasks in Ludwig 0.10.1.">
+                    <option value="loss" selected="true">Loss</option>
+                </param>
```
[the remark this answers is not in the document]
Comment (paulocilasjr) [reply tools/imagelearner/image_learner.xml]: I agree. 
XML was modified to restore the `_resolve_validation_metric` function when the user does not select a specific metric.
